### DISCUSSION TITLE
Parallelized implementation of DIA-NN pipeline

### DISCUSSION THREAD

**Discussion by Munchic** [vdemichev/DiaNN#446] [Q&A]
Hi Vadim,

I came across this thread: https://github.com/bigbio/quantms/issues/164 and am implementing a similar step-by-step pipeline for analyzing phosphosites. I wanted to gather your feedback on whether my implementation is correct, thank you so much!

### Steps
1. Spectral library generation (with an underlying DDA library generated by FragPipe/easypqp)
2. Parallelized first pass of each of the runs (Thermo DIA)
3. Cross-analysis and second pass of all files together

### Code  
**Step 1**
```
diann-1.8.1 --lib $lib --threads ${num_cpus} --verbose 1 \
      --out "out/report.tsv" --qvalue 0.01 --matrices --temp "out" --out-lib "out/speclib.tsv" --gen_spec_lib --prosit --predictor --fasta $db --fasta_search \
      --min-fr-mz 300 --max-fr-mz 1400 --met_excision --cut K*,R* --missed-cleavages 1 \
      --min-pep-len 7 --max-pep-len 30 --min-pr-mz 400 --max-pr-mz 1000 --min-pr-charge 1 --max-pr-charge 4 \
      --unimod4 --var-mods 2 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --var-mod UniMod:21,79.966331,STY \
      --use-quant --reanalyse --smart-profiling --peak-center --no-ifs-removal --global-norm
```
where `$lib` is the DDA .tsv library and `$db` is the FASTA for human proteins

Log: [step1_log.txt](https://github.com/vdemichev/DiaNN/files/9181972/step1_log.txt)

**Step 2**
Spins up multiple different instances that each process a single DIA run
```
diann-1.8.1 --dir "data" --lib $lib --threads ${num_cpus} --verbose 1 \
      --out "out/report.tsv" --out-lib "out/spect_lib.tsv" --qvalue 0.01 --matrices --temp "data" $predictor --prosit --rt-profiling  \
      --min-fr-mz 300 --max-fr-mz 1400 --met_excision --cut K*,R* --missed-cleavages 1 \
      --min-pep-len 7 --max-pep-len 30 --min-pr-mz 400 --max-pr-mz 1000 --min-pr-charge 1 --max-pr-charge 4 \
      --unimod4 --var-mods 2 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --var-mod UniMod:21,79.966331,STY \
      --smart-profiling --peak-center --no-ifs-removal --global-norm --monitor-mod UniMod:21
```
where `$lib` is the generated `.predicted.speclib` library from Step 1

Log: [step2_log.txt](https://github.com/vdemichev/DiaNN/files/9182037/step2_log.txt)

**Step 3**
All .quant and .mzML files from the previous step are gathered in the `data` folder:
```
diann-1.8.1 --dir "data" --lib $lib --threads ${num_cpus} --verbose 1 \
      --out "out/report.tsv" --out-lib "out/spect_lib.tsv" --qvalue 0.01 --matrices --temp "data" $predictor --prosit --rt-profiling \
      --min-fr-mz 300 --max-fr-mz 1400 --met_excision --cut K*,R* --missed-cleavages 1 \
      --min-pep-len 7 --max-pep-len 30 --min-pr-mz 400 --max-pr-mz 1000 --min-pr-charge 1 --max-pr-charge 4 \
      --unimod4 --var-mods 2 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --var-mod UniMod:21,79.966331,STY \
      --use-quant --reanalyse --smart-profiling --peak-center --no-ifs-removal --global-norm --monitor-mod UniMod:21
```
where `$lib` is the generated `.predicted.speclib` library from Step 1

Log: [step3_log.txt](https://github.com/vdemichev/DiaNN/files/9182064/step3_log.txt)

Thank you so much for your help and let me know if there are any changes I should make or if I can provide more information. Thank you so much!

Sincerely,
Munchic

**Comment by vdemichev** [comment]
Hi Munchic,

Some comments:
- Limiting precursor charge range to 2-3 will reduce search space
- If the library is not expected to include almost all of the phosphopeptides, makes sense to enable up to 3 var mods
- Ox(M) is likely detrimental

Best,
Vadim

**Comment by Munchic** [comment]
Hi Vadim,

Thank you so much for your response!

- Got it, thanks! In your experience, does this reduce IDs for a similar search?
- Do you mean if the DDA library does not include all phosphopeptides, I should expand to 3 var mods? Upon looking at the outputs of 2 and 3 var mods, we saw that the number of unique phosphopeptides identified was slightly higher in 2 var mods. Is this expected?
- Could you clarify in what sense Ox(M) is detrimental -- in terms of performance or search results?

**Reply by vdemichev** [reply]
- No or 1-2% loss of IDs in most cases
- If DDA library is super-comprehensive already, but does not have all peptides (e.g. 10% of peptides are likely to come from FASTA), then makes sense to keep 2 var mods. If e.g. 50% of peptides will come from FASTA - might make sense to switch to 3 var mods. This will also depend on what kind of enrichment you use. If in your tests 2 is better, then it's good, smaller search space is faster.
- Spearch space expansion => lower confidence & slower search.

**Reply by Munchic** [reply]
In our case, the DDA library specifies only a small subset of peptides, so most will come from the FASTA file. At the same time, during search it doesn't seem like there are any peptides with 3 phosphorylated sites. So I guess, smaller search space with 2 var mods would work here?

Understood, thank you so much for your feedback. Very appreciated! 🙂

**Comment by Munchic** [comment]
Otherwise, in terms of the command usage, are the steps correctly implemented? Thank you.

**Reply by vdemichev** [reply]
Seems OK, but I haven't checked all the commands

**Reply by Munchic** [reply]
Mainly, I wanted to check if it's correct to use `--use-quant --reanalyse --rt-profiling` in Step 3 to analyze the results from Step 2 together?

**Reply by vdemichev** [reply]
Yes, all good

**Reply by Munchic** [reply]
Thank you, Vadim!
